R/mixture_model_wrapper.R

```diff
@@ -0,0 +1,175 @@
+#' `function(...){EpiSoon::fable_model(model = , ...)}` or
+#' `function(...){EpiSoon::bsts_model(model = , ...)}`.
+#' @inheritParams bsts_model
+#' @param weighting_period The number of most recent timepoints to hold out to
+#' generate the weights for the mixture model
+#' @return A dataframe of predictions (with columns representing the
+#'  time horizon and rows representing samples).
+#' @export
+#' @importFrom purrr map_dfr
+#' @importFrom dplyr bind_rows n ungroup mutate rename group_by filter
+#' @importFrom tidyr pivot_longer pivot_wider
+#' @examples
+#'
+#' y <- EpiSoon::example_obs_rts[1:10, ]$rt
+#' model = fable::ARIMA(y ~ time)
+#' samples = 10
+#' horizon = 7
+#' weighting_period = 5
+#'
+#' # make list with models
+#' baseline_ensemble <- function(...) {
+#' EpiSoon::fable_model(model = fabletools::combination_model(fable::ARIMA(y), fable::ETS(y), fable::NAIVE(y),
+#'                                                            fable::RW(y ~ drift()), cmbn_args = list(weights = "inv_var")), ...)
+#' }
+#' models <- list(
+#'   Mixfabl = baseline_ensemble,
+#'   "ARIMA" = function(...){EpiSoon::fable_model(model = fable::ARIMA(y), ...)},
+#'   "ETS" = function(...){EpiSoon::fable_model(model = fable::ETS(y), ...)},
+#'   "Drift" = function(...){EpiSoon::fable_model(model = fable::RW(y ~ drift()), ...)}
+#' )
+#'
+#' # make forecst on its own
+#' forecast <- crps_ensemble(y = y,
+#'                           models = models,
+#'                           samples = 10,
+#'                           horizon = 7,
+#'                           weighting_period = 5)
+#'
+#'
+#' # together with forecast_rt
+#' fc_rt <- forecast_rt(EpiSoon::example_obs_rts[1:10, ],
+#'             model = function(...){
+#'               crps_ensemble(models = models,
+#'                             weighting_period = 5,
+#'                             ...)},
+#'             samples = 10,
+#'             horizon = 7)
+#'
+#' forecast_eval <- evaluate_model(EpiSoon::example_obs_rts,
+#'                EpiSoon::example_obs_cases,
+#'                model = function(...){
+#'                 crps_ensemble(models = models,
+#'                             weighting_period = 5,
+#'                             ...)},
+#'                horizon = 7, samples = 10,
+#'                serial_interval = example_serial_interval,
+#'                min_points = 10)
+#'
+#' plot_forecast_evaluation(forecast_eval$forecast_rts,
+#'                          EpiSoon::example_obs_rts,
+#'                          horizon_to_plot = 7)
+#'
+#'
+
+
+crps_ensemble <- function(y = NULL,
+                          models = NULL,
+                          samples = NULL,
+                          horizon = NULL,
+                          weighting_period = 5) {
+
+
+  #### Error Handling
+  # check if y is there
+  if(is.null(y)) stop("parameter y is missing")
+
+
+  # check if stackr is installed
+  if (!suppressWarnings(require("stackr", quietly = TRUE) == TRUE)) {
```

Comment:
Can you replace this with the check suggest function implemented for `brms`

Reply:
I did. For now I copied the function to change the custom message so people can find the repo.

Reply:
Suggest you add an arguement to check_suggests that tells it that suggestion is a devtools package and what to do to install it. Something like 

```
check_suggests("stackr", dev_message = "Install using devtools::install_githuwgwegwegwe") 

Then have and if (!missing(dev_message)) {
Custom message for dev package
}else{
Current install from CRAN message
}

Reply:
that is nice!